packages/core/src/common/messaging/proxy-factory.ts

```diff
@@ -132,14 +132,8 @@ export class JsonRpcProxyFactory<T extends object> implements ProxyHandler<T> {
      * response.
      */
     listen(connection: MessageConnection): void {
-        if (this.target) {
```

Comment:
Can you move this check to `onRequest` and `onNotification` just to make the error more clear once it fails there? With this change we'll get a weird error I feel, not sure.

Reply:
I left it out on purpose to allow pass proxies which cannot enumerate own properties.

Reply:
I meant `this.target` only :)

Before it used to not call `onRequest` if the target wasn't set, but now it will execute and try to access something on an `undefined` value: `this.target[method](...args)`. I did not try this kind of corner case but I could forsee it happening hence why I asked.